File: leetcode/rust-leetcode/src/2134.minimum-swaps-to-group-all-1s-together-ii.rs
impl Solution {
    pub fn min_swaps(nums: Vec<i32>) -> i32 {
        let ones = nums.iter().sum::<i32>() as usize;
        let (mut ans, mut zeros) = (nums.len()-ones, 0);
        for i in 0..ones {
            if nums[i] & 1 == 0 {
                zeros += 1;
            }
        }
        ans = ans.min(zeros);
        for i in ones..nums.len()+ones {
            if nums[i-ones] & 1 == 0 {
                zeros -= 1;
            }
            if nums[i % nums.len()] & 1 == 0 {
                zeros += 1;
            }
            ans = ans.min(zeros);
        }
        ans as _
    }
}

File: leetcode/go/2134.minimum-swaps-to-group-all-1s-together-ii.go
func minSwaps(nums []int) int {
    l := len(nums)
    ones, zeros := 0, 0
    for _, n := range nums {
        if n == 1 {
            ones++
        }
    }
    for i:=0; i<ones;i++ {
        if nums[i] == 0 {
            zeros++
        }
    }
    ans := min(l-ones, zeros)
    for i:=ones; i<l+ones; i++ {
        if nums[i - ones] == 0 {
            zeros--
        }
        if nums[i % l] == 0 {
            zeros++
        }
        ans = min(ans, zeros)
    }
    return ans
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

